src/realm/object-store/shared_realm.cpp

```diff
@@ -1087,9 +1089,19 @@ void Realm::convert(const Config& config, bool merge_into_existing)
     verify_thread();
 
 #if REALM_ENABLE_SYNC
-    if (config.sync_config && config.sync_config->flx_sync_requested) {
-        throw std::logic_error("Realm cannot be converted if flexible sync is enabled");
+    auto src_is_flx_sync = m_config.sync_config && m_config.sync_config->flx_sync_requested;
+    auto dst_is_flx_sync = config.sync_config && config.sync_config->flx_sync_requested;
+    auto dst_is_pbs_sync = config.sync_config && !config.sync_config->flx_sync_requested;
+
+    if (dst_is_flx_sync && !src_is_flx_sync) {
+        throw std::logic_error(
+            "Realm cannot be converted to a flexible sync realm unless flexible sync is already enabled");
+    }
+    if (dst_is_pbs_sync && src_is_flx_sync) {
+        throw std::logic_error(
+            "Realm cannot be converted from a flexible sync realm to a partition based sync realm");
     }
```

Comment:
So it looks like the following conversions are allowed. Correct?
FLX -> Local (tested below)
FLX -> FLX (tested below)
PBS -> Local
PBS -> PBS
Local -> Local
Local -> PBS

Reply:
Correct. The other conversions are tested [here](https://github.com/realm/realm-core/blob/a9cafd56734ea25478d9deae153ded23865cec4b/test/object-store/realm.cpp#L1021).